util/versions/version.go

```diff
@@ -1,4 +1,4 @@
-package helm
+package versions
 
 import "github.com/Masterminds/semver/v3"
 
```

Comment:
I reviewed this in my OCI branch, and I think something that would be useful would be a `IsConstraint` function.

```
func IsConstraint(text string) bool {
	_, err := semver.NewConstraint(text)
	return err == nil
}
```

Reply:
That sounds good - but for the sake of our logic, I'd recommend ensuring that a constraint can never be a valid version (which is the root of this whole issue!)

```
func IsConstraint(text string) bool {
	_, err := semver.NewConstraint(text)
	return !IsVersion(text) && err == nil
}
```